### PR TITLE
Ft user oauth google and facebook

### PR DESCRIPTION
**Pull request by gracefideledushime**
##What does this PR do?
create a new branch for user login by google oauth and facebook

##Screenshots
N/A

### CODE REVIEW

**Review by kelliaUmuhire — APPROVED** [review]
Well done!

**Review by nbrucepro — APPROVED** [review]

**Review by istafrica-dennw — CHANGES_REQUESTED** [review]
You need to write tests for your feature

File: .eslintrc.json [hunk 1]
```diff
@@ -1,3 +1,57 @@
+<<<<<<< HEAD
```
Comment (leny62) [inline .eslintrc.json]: @Graziee6 please remove such trailing spaces

File: .eslintrc.json [hunk 2]
```diff
@@ -1,3 +1,57 @@
...
+    //             "FunctionDeclaration": true,
+    //             "MethodDefinition": true,
+    //             "ClassDeclaration": true
+    //         }
+    //     }
+    // ]
+  },
+  "quotes": [2, "single", "avoid-escape"]
+}
+=======
```
Comment (leny62) [inline .eslintrc.json]: @Graziee6 , your files have trailing spaces

**Review by leny62 — CHANGES_REQUESTED** [review]
@Graziee6 , your tests are not passing. And one of the reasons is the trailing spaces you haven't removed

**Comment by leny62** [comment]
@Graziee6 , please rebase!

**Review by leny62 — APPROVED** [review]

**Pull request merged**